### PR TITLE
blockmanager: Avoid duplicate header announcements.

### PR DESCRIPTION
**Pull request by davecgh**
Clients are able to choose between two modes of notification about new blocks via the wire protocol.  The first, and current default, is via inv messages and the second is via direct header announcements.

There is code in the peer package which deals with filtering duplicate inventory announcements via the `QueueInventory` message.  This works as expected for inv-based announcements, however, direct header announcements are not inv messages and hence do not use that code path.

Consequently, the recent addition of the fast relay path which relied on that filtering to avoid sending duplicate announcements resulted in duplicate announcements being sent to peers who requested the header announcements mode.

This resolves the issue by adding some logic directly to the notifications themselves to deal with filtering duplicates, both for inv and header-based announcements.

### CODE REVIEW

**Comment by matheusd** [comment]
Should note that the fix itself worked on my tests, so the previous comment was just a clarification.

**Comment by davecgh** [comment]
As the name `NTNewTipBlockChecked` indicates, it is only called when a new tip block is connected. We still need to relay side chain blocks.

**Comment by alexlyp** [comment]
tACK on testnet

File: blockmanager.go [hunk 1]
```diff
@@ -330,6 +330,10 @@ type blockManager struct {
 	cachedCurrentTemplate *BlockTemplate
 	cachedParentTemplate  *BlockTemplate
 	AggressiveMining      bool
+
+	// The follownig field are used to filter duplicate block announcements.
```
Comment (jrick) [inline blockmanager.go]: typo: follownig

Also use fields instead of field?

Comment (davecgh) [reply blockmanager.go]: Updated.

**Pull request merged**